tests/kernel/interrupt/CMakeLists.txt

```diff
@@ -15,5 +15,8 @@ target_sources(app PRIVATE
     src/nested_irq.c
     )
 
+if(CONFIG_CPU_CORTEX_M)
+  target_sources_ifdef(CONFIG_ISR_SHARE_IRQ app PRIVATE src/shared_irq.c)
+endif()
```

Comment:
This is not the correct way to implement this. See my later comment.

Reply:
The guard has been removed.